incubator/binding-kafka-grpc/src/main/java/io/aklivity/zilla/runtime/binding/kafka/grpc/internal/stream/KafkaGrpcRemoteServerFactory.java

```diff
@@ -1833,16 +1828,7 @@ private void onKafkaStreamInitializationSignal(
 
         final long traceId = supplyTraceId.getAsLong();
 
-        KafkaGrpcBindingConfig binding = bindings.get(lastBindingId);
-
-        binding.routes.forEach(r ->
-            r.when.forEach(c ->
-            {
-                KafkaGrpcConditionResult result = c.resolve();
-                RemoteServer remoteServer =
-                    new RemoteServer(binding.id, r.id, binding.options.entryId, result, binding.helper);
-                remoteServer.fetch.doKafkaBegin(traceId, 0L, 0L);
-            }));
+        kafkaRemoteServers.forEach(r -> r.initiate(traceId));
```

Comment:
Suggest renaming to `servers`.

Lambda should be `s -> s.initiate(traceId)` (not `r`).